packages/m2m-gateway/test/api/agreements/suspendAgreement.test.ts

```diff
@@ -0,0 +1,82 @@
+import { describe, it, expect, vi } from "vitest";
+import { generateToken } from "pagopa-interop-commons-test";
+import { AuthRole, authRole } from "pagopa-interop-commons";
+import request from "supertest";
+import { m2mGatewayApi } from "pagopa-interop-api-clients";
+import { api, mockAgreementService } from "../../vitest.api.setup.js";
+import { appBasePath } from "../../../src/config/appBasePath.js";
+import {
+  missingMetadata,
+  resourcePollingTimeout,
+} from "../../../src/model/errors.js";
+import { getMockedApiAgreement } from "../../mockUtils.js";
+import { toM2MGatewayApiAgreement } from "../../../src/api/agreementApiConverter.js";
+
+describe("POST /agreements/:agreementId/suspend router test", () => {
+  const mockApiAgreement = getMockedApiAgreement();
+
+  const mockM2MAgreementResponse: m2mGatewayApi.Agreement =
+    toM2MGatewayApiAgreement(mockApiAgreement.data);
+
+  const makeRequest = async (
+    token: string,
+    agreementId: string = mockApiAgreement.data.id
```

Comment:
```suggestion
    agreementId: AgreementID = mockApiAgreement.data.id
```
Same for the other tests